io-engine/src/grpc/v0/nexus_grpc.rs

```diff
@@ -38,14 +38,9 @@ impl From<&NexusChild<'_>> for rpc::ChildState {
             ChildState::Destroying => rpc::ChildState::ChildDegraded,
             ChildState::Closed => rpc::ChildState::ChildDegraded,
             ChildState::Faulted(reason) => match reason {
-                Reason::NoSpace => rpc::ChildState::ChildDegraded,
-                Reason::TimedOut => rpc::ChildState::ChildDegraded,
-                Reason::Unknown => rpc::ChildState::ChildFaulted,
-                Reason::CantOpen => rpc::ChildState::ChildFaulted,
-                Reason::RebuildFailed => rpc::ChildState::ChildFaulted,
-                Reason::IoError => rpc::ChildState::ChildFaulted,
-                Reason::ByClient => rpc::ChildState::ChildFaulted,
-                Reason::AdminCommandFailed => rpc::ChildState::ChildFaulted,
+                FaultReason::IoError => rpc::ChildState::ChildFaulted,
+                s if s.is_recoverable() => rpc::ChildState::ChildDegraded,
```

Comment:
How does control-plane know when it can/has to online a Child in order to make it healthy again?
We should document this well, and in the protobuf too, otherwise we risk a child being degraded forever, because common control-plane logic is to leave degraded children alone.

Reply:
Will do on this ticket: https://dcsw-bolt.atlassian.net/browse/GTM-469